### ISSUE TITLE
Overhaul Weather System

### ISSUE THREAD

**Issue by Qerbas** [NeoArmageddon/co10_Escape#20]
For some time the weather system has not worked properly, which reduces a lot of the variety that the Escape missions are known for along with the improved ambiance that different weather settings add.

Firstly, there doesn't seem to be much, if any, difference between the Rain and Storm settings, and all settings seem to vary based on map.  For example, some maps like Altis always seem to have rain at the beginning when the "Rain" setting is on, however Stratis almost never does, nor does rain ever seem to actually start.  Fog is especially different between different maps.

More importantly, any setting with cloud cover doesn't seem to be using the forceweatherchange command properly because when initially spawning on a map with the Rain, Storm, or Overcast setting, the sky will _always_ look like this: 
![Rain-Constant-Stale- Vars-Altis](https://user-images.githubusercontent.com/53233871/63718995-c8724d00-c800-11e9-8a09-a3c4b6664d4c.png)

It will stay that way until the player switches their cloud settings from their current setting to any other setting in the video options and then the clouds will render properly.

The forceweatherchange command must be applied at the wrong time or in the wrong way because in my testing using a test mission with just the lines `0 setovercast x; forceweatherchange;` (where x is any value between 0 and 1) in initserver.sqf the clouds always rendered properly for all clients on a dedicated server.

Not only that, but the weather settings are not properly overriding the weather parameters set in the mission.sqm because on Tembelan Island, the map spawns with extremely thick fog around 90% of the time no matter what weather parameters are selected.  I had to remove the `startFogBase=250;` line from the mission.sqm to fix that issue.  This is most likely the same issue that is causing the inconsistency between maps that I noticed with the Rain setting.

These issues make it so that functionally escape can really only be played with the "Clear" setting because it seems to be the only parameter that works close to how it's intended.  It's a shame because Arma 3's volumetric clouds look really nice and the sound effects, especially for lightning are excellent.

I've done some research on the weather parameters and it seems that gradually shifting weather may not be possible without incredibly long time scales.  It may just be more efficient for now just to have fixed weather options until feasibility of dynamic weather can be determined.  I'll do some more testing on that.  At least at the moment, the current dynamic weather system is effectively broken.

So for now, I think there should just be fixed lobby parameters and the settings I would recommend as options would be Clear, Light Clouds, Medium Clouds, Overcast, Rain, Storm, Light Fog, Medium Fog, Heavy Fog, Light Fog with Rain, Medium Fog with Rain, Heavy Fog with Rain, and Heavy Fog with Storm.  The main difference between "Rain" and "Storm" would be Storm would have the lightning values set to max and wind set to a high value.  All values should be set manually by script (except for fogbase and fogdecay which is better set on a per map basis in the editor) with the "Manual Override" option set for Rain, Lightnings, Waves, and Winds to ensure that Arma doesn't override the scripted values.  It's especially important for [lightning](https://community.bistudio.com/wiki/setLightnings) and [waves](https://community.bistudio.com/wiki/setWaves), since those settings don't work at all unless the Manual Override option is selected.

**Comment by Qerbas** [comment]
Most recent changes to the weather system fix the issue where the sky looks like my screenshot when on a setting with full overcast.  A cursory check shows that the settings seem to work for the most part, however setting "rain" or "storm" just makes full overcast and rain never seems to start.  I did get light rain one time with I think when I had the weather and weather trend set to random.  Will need to do full sessions to see how well it works in real world conditions since Arma's weather can't change in short time periods, at least for clouds if the wiki page for [setOvercast](https://community.bistudio.com/wiki/setOvercast) is correct.

Tembelan Island issue still occurs, but the edit I just pushed fixes that.

**Comment by NeoArmageddon** [comment]
Rain takes several tenth of minutes more compared to overcast to actually starts.
Maybe switching from a dynamic weather system to a static, inengine weather and forecast system is better.

**Comment by Qerbas** [comment]
I definitely agree.  Scripted weather just doesn't seem worth the hassle.  Having a wide variety of parameters with static weather should be an overall improvement to the gameplay experience.

I did some more testing and it turns out the weather is still very inconsistent, especially the fog setting.  It even happens when when testing on maps that don't have the startfogbase and forcastfogbase lines.  The fog is very thin and the cloud cover isn't thick enough which looks strange.

Here's an example of that: 
![ArmA 3 Screenshot 2019 11 10 - 04 46 55 96](https://user-images.githubusercontent.com/53233871/68544179-575d2300-0375-11ea-866e-61581def419c.png)

**Comment by Qerbas** [comment]
So, I've been playing with the weather system on for a while now and for the most part it works well.  The clouds always render properly and there's a good bit of variety over the course of a normal mission.  Sometimes clouds, sometimes storms, usually pretty clear.  The main thing I've noticed is that some of the startup weather parameters don't seem to work properly. 

Here's my notes from numerous tests where I load up escape and look around for a couple minutes.

Dedicated server
Clear/Random:  Seems to work very consistently
Overcast/Random:  seems to only produce light clouds at the most
Rain/Random:  inconsistent.  produces full overcast sometimes but partial clouds other times.  not sure if rain ever arrives.
Fog/Random:  seems to consistently produce light fog with heavier fog rarely. clouds sometimes spawn too varying from none to heavy overcast.
Storm/Random:  produces rain occasionally, but mostly just produces heavy overcast.  seems to just produce light clouds fairly frequently
Random/Random: seems to work as expected

Clear/Constant:  seems to work very consistently
Overcast/Constant:  seems to work pretty consistently with medium clouds
Rain/Constant:  consistently produces overcast, but never seems to produce rain
Fog/Constant:  similar to Fog/Random
Storm/Constant:  seems to produce heavy overcast consistently but never rain
Random/Constant: seems to work as expected

Having those starting parameters working properly would be good.  I took a look at fn_weather.sqf, but I can't quite figure out how it works.  I tried deleting all but one of the `_weathertemplates pushbBack [x,x,x,x]` and tried adjusting the values, but only the first and third numbers in the array worked, which controls overcast and fog respectively.  The others didn't do anything.  Setting the second to 1 should in theory produce full rain right at the outset but it doesn't.
